### PR TITLE
Feat/paypro

### PR DESCRIPTION
**Pull request by matiu**
Adds payment protocol support

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -757,9 +798,14 @@ API.prototype.signTxProposalFromAirGapped = function(txp, encryptedPkr, m, n) {
   self.credentials.n = n;
   self.credentials.addPublicKeyRing(publicKeyRing);
 
-  if (!Verifier.checkTxProposal(self.credentials, txp)) {
-    throw new Error('Fake transaction proposal');
-  }
+  // When forAirGapped=true -> checkTxProposal is sync
```
Comment (isocolsky) [inline lib/api.js]: Making the `signTxProposalFromAirGapped` method async is the right thing to do IMO.

**Comment by isocolsky** [comment]
Great work!

File: lib/api.js (hunk 1 above)
Comment (matiu) [inline lib/api.js]: ACK, will do.

On Thu, Mar 26, 2015 at 10:46 PM, Ivan Socolsky <notifications@github.com>
wrote:

> In lib/api.js
> <https://github.com/bitpay/bitcore-wallet-client/pull/43#discussion_r27271622>
> :
>
> > @@ -757,9 +798,14 @@ API.prototype.signTxProposalFromAirGapped = function(txp, encryptedPkr, m, n) {
> >    self.credentials.n = n;
> >    self.credentials.addPublicKeyRing(publicKeyRing);
> >
> > -  if (!Verifier.checkTxProposal(self.credentials, txp)) {
> > -    throw new Error('Fake transaction proposal');
> > -  }
> > +  // When forAirGapped=true -> checkTxProposal is sync
>
> Making the signTxProposalFromAirGapped method async is the right thing to
> do IMO.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore-wallet-client/pull/43/files#r27271622>.
>



-- 
Matías Alejo Garcia
@ematiu
Roads? Where we're going, we don't need roads!

**Comment by matiu** [comment]
updated

**Comment by matiu** [comment]
Working on coverage now.

**Pull request merged**